org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/StateBasedContainer.java

```diff
@@ -110,7 +115,8 @@ public IResourceDescription getResourceDescription(URI uri) {
 	public Iterable<IEObjectDescription> getExportedObjectsByType(EClass type) {
```

Comment:
Given that it is easier to override / provide a custom container, do we really need the changes in this class exception for the newly introduced getter?

Reply:
the idea of that whole change was to solve the problem at its cause, not giving the user an easier way solving the issue on his own.

acutally the fix in our codebase looks like:
`class FilteringResourceDescriptionStateBaseContainerManager extends StateBasedContainerManager {

	@FinalFieldsConstructor
	static class FilteringResourceDescriptions implements IResourceDescriptions {
		val IResourceDescriptions delegate
		val Predicate<IResourceDescription> predicate

		override getAllResourceDescriptions() {
			delegate.allResourceDescriptions.filter[predicate.test(it)]
		}

		override getResourceDescription(URI uri) {
			val res = delegate.getResourceDescription(uri)
			
			if(res === null) return null
			
			if (predicate.test(res)) {
				return res
			}
			return null
		}

		override getExportedObjects() {
			return allResourceDescriptions.map[exportedObjects].flatten
		}

		override getExportedObjects(EClass type, QualifiedName name, boolean ignoreCase) {
			return allResourceDescriptions.map[getExportedObjects(type, name, ignoreCase)].flatten
		}

		override getExportedObjectsByObject(EObject object) {
			return allResourceDescriptions.map[getExportedObjectsByObject(object)].flatten
		}

		override getExportedObjectsByType(EClass type) {
			return allResourceDescriptions.map[getExportedObjectsByType(type)].flatten
		}

		override isEmpty() {
			return getExportedObjects.isEmpty
		}
	}

	// copied from super. created a pull request for xtext https://github.com/eclipse/xtext-core/pull/1253
	override protected createContainer(String handle, IResourceDescriptions resourceDescriptions) {
		val globalState = getState(resourceDescriptions)

		// actual change
		val filteringDescriptions = new FilteringResourceDescriptions(resourceDescriptions) [
			globalState.containsURI(handle, it.URI)
		]
		var IContainerState containerState = new ContainerState(handle, globalState)
		var StateBasedContainer result = new StateBasedContainer(filteringDescriptions, containerState)
		// end
		if(globalState instanceof FlatResourceSetBasedAllContainersState) result.setUriToDescriptionCacheEnabled(false)
		return result
	}

}

class ContainerState implements IContainerState {
	final String root
	final IAllContainersState globalState

	protected new(String root, IAllContainersState globalState) {
		this.root = root
		this.globalState = globalState
	}

	override Collection<URI> getContents() {
		return globalState.getContainedURIs(root)
	}

	override boolean contains(URI uri) {
		return globalState.containsURI(root, uri)
	}

	override boolean isEmpty() {
		return globalState.isEmpty(root)
	}

	override String toString() {
		return '''container '«»«root»' from «globalState»'''
	}
}`